cargo-near-build/src/types/near/build/input/docker_context.rs

```diff
@@ -14,100 +14,183 @@ impl super::Opts {
         Ok(contract_path)
     }
 
-    const BUILD_COMMAND_CLI_CONFIG_ERR: &'static str =  "cannot be used, when `container_build_command` is configured from `[package.metadata.near.reproducible_build]` in Cargo.toml";
+    const BUILD_COMMAND_CLI_CONFIG_ERR: &'static str =  "flag cannot be used, when `container_build_command` is configured from `[package.metadata.near.reproducible_build]` in Cargo.toml";
 
     pub fn get_cli_build_command_in_docker(
         &self,
         manifest_command: Option<Vec<String>>,
-    ) -> eyre::Result<String> {
-        if let Some(manifest_command) = manifest_command {
-            // NOTE: `--no-locked` is allowed for docker builds
-            // if self.no_locked {
-            //     no-op
-            // }
-            if self.no_release {
-                return Err(eyre::eyre!(format!(
-                    "`{}` {}",
-                    "--no-release",
-                    Self::BUILD_COMMAND_CLI_CONFIG_ERR
-                )));
-            }
-            if self.no_abi {
-                return Err(eyre::eyre!(format!(
-                    "`{}` {}",
-                    "--no-abi",
-                    Self::BUILD_COMMAND_CLI_CONFIG_ERR
-                )));
-            }
-            if self.no_embed_abi {
-                return Err(eyre::eyre!(format!(
-                    "`{}` {}",
-                    "--no-embed-abi",
-                    Self::BUILD_COMMAND_CLI_CONFIG_ERR
-                )));
-            }
-            if self.no_doc {
-                return Err(eyre::eyre!(format!(
-                    "`{}` {}",
-                    "--no-doc",
-                    Self::BUILD_COMMAND_CLI_CONFIG_ERR
-                )));
-            }
-            if self.features.is_some() {
-                return Err(eyre::eyre!(format!(
-                    "`{}` {}",
-                    "--features",
-                    Self::BUILD_COMMAND_CLI_CONFIG_ERR
-                )));
-            }
-            if self.no_default_features {
-                return Err(eyre::eyre!(format!(
-                    "`{}` {}",
-                    "--no-default-features",
-                    Self::BUILD_COMMAND_CLI_CONFIG_ERR
-                )));
+        passed_env: Option<Vec<String>>,
+    ) -> eyre::Result<Vec<String>> {
+        manifest_command.map_or_else(
+            || {
+                println!(
+                    "{}",
+                    "configuring `container_build_command` from cli args, passed to current command".cyan()
+                );
+                println!();
+                Ok(self.passthrough_some_opts_into_docker_cmd())
+            },
+            |mut manifest_command| {
+                self.check_flag_conflicts_with_manifest_command()?;
+                let suffix_env = passed_env.into_iter().flatten().filter(|env_key| {
+                    std::env::var(env_key).is_ok()
+                }).flat_map(|env_key| {
```

Comment:
nit: It is really hard to reason about. Simple `if let` would be far easier to read, IMO